tests/split-test.toit

```diff
@@ -0,0 +1,53 @@
+// Copyright (C) 2023 Toitware ApS.
+// Use of this source code is governed by a Zero-Clause BSD license that can
+// be found in the tests/TESTS_LICENSE file.
+
+import expect show *
+import fs
+import fs.posix
+import fs.windows
+import system
+
+POSIX-TESTS ::= [
+  [[], ""],
+  [["/"], "/"],
+  [["a"], "a"],
+  [["a", "b"], "a/b"],
+  [["/", "a"], "/a"],
+  [["/", "a", "b"], "/a/b"],
+  [["/", "a", "b"], "/a////b"],
+  [["/"], "/"],
+  [["/", "a"], "//a"],
+]
+
+WINDOWS-TESTS ::= [
+  [["\\", "foo"], "\\foo"],
+  [[], ""],
+  [["\\"], "/"],
+  [["a"], "a"],
+  [["a", "b"], "a\\b"],
+  [["\\", "a"], "\\a"],
+  [["\\", "a", "b"], "\\a\\b"],
+  [["directory", "file"], "directory\\file"],
+  [["C:\\", "Windows", "System32"], "C:\\Windows\\System32"],
+  [["C:\\", "Windows"], "C:\\Windows"],
+  [["C:", "a"], "C:a"],
+  [["C:", "a", "b"], "C:a\\b"],
+  [["C:\\", "a"], "C:\\a"],
+  [["\\\\host", "share", "foo"], "\\\\host\\share\\foo"],
```

Comment:
Not sure I agree with this.
Shouldn't we just split the volume-name entirely in its own segment, even if it contains separators?